src/types.ts

```diff
@@ -140,27 +198,20 @@ export interface IEmailAddress {
   email: string;
 }
 
-export interface IThreadEmail {
-  id: string;
-  mailboxIds: string[];
-  isUnread: boolean;
-  isFlagged: boolean;
-}
-
-export interface IThread {
+/**
+ * See https://jmap.io/spec-mail.html#threads
+ */
+export interface IThreadProperties {
   id: string;
-  emails: IThreadEmail[];
-  createdModSeq: number;
-  updatedModSeq: number;
-  deleted: Date | null;
+  emailsIds: string[];
 }
 
 export interface IMailbox extends IMailboxProperties, IMailboxData {}
```

Comment:
IMailboxData seems unused, it comes from https://jmap.io/server.html#3-mailboxes and is outdated.